cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/PreferredLeaderElectionGoal.java

```diff
@@ -63,21 +64,38 @@ private void sanityCheckOptimizationOptions(OptimizationOptions optimizationOpti
   }
 
   private void maybeMoveReplicaToEndOfReplicaList(Replica replica, ClusterModel clusterModel) {
-    // There are two scenarios where replica swap operation is skipped:
+    // There are three scenarios where replica swap operation is skipped:
     // 1.the replica is not leader replica and _excludeFollowerDemotion is true.
     // 2.the replica's partition is currently under replicated and _skipUrpDemotion is true.
-    if (!(_skipUrpDemotion && isPartitionUnderReplicated(_kafkaCluster, replica.topicPartition()))
+    // 3.the replica doesn't exist.
+    boolean skipUrp = false;
```

Comment:
The name skipUrp is kind of misleading here. We should probably name it as "skipReplicaMove"

Reply:
Done